Objects/stringlib/eq.h

```diff
@@ -4,16 +4,8 @@
  * unicode_eq() is called when the hash of two unicode objects is equal.
  */
 Py_LOCAL_INLINE(int)
-unicode_eq(PyObject *aa, PyObject *bb)
+unicode_eq(PyObject *a, PyObject *b)
 {
-    assert(PyUnicode_Check(aa));
-    assert(PyUnicode_Check(bb));
-    assert(PyUnicode_IS_READY(aa));
-    assert(PyUnicode_IS_READY(bb));
```

Comment:
Note to self: AFAICS, all of these are covered in callers of `unicode_eq`, so this should be ok.

Reply:
PyUnicode_GET_LENGTH() and other macros *do* check that (1) the object is an unicode string (2) the string is ready. These assertions were already there in Python 3.10:

```
#define PyUnicode_GET_LENGTH(op)                \
    (assert(PyUnicode_Check(op)),               \
     assert(PyUnicode_IS_READY(op)),            \
     ((PyASCIIObject *)(op))->length)
```

Note: Inada-san just removed the concept of "ready" string by removing PyUnicode_WCHAR_KIND (PEP 623).